test/test_autograd.py

```diff
@@ -1618,20 +1611,22 @@ def test_reentrant(self):
 
         class Reenter(Function):
             @staticmethod
-            def forward(ctx, x_data):
-                ctx.x = Variable(x_data, requires_grad=True)
-                ctx.y = Variable(y_data, requires_grad=True)
-                ctx.output_var = ctx.x * ctx.y
-                return ctx.output_var.data
+            def forward(ctx, x):
+                with torch.enable_grad():
+                    ctx.x = Variable(x.data, requires_grad=True)
+                    ctx.y = Variable(y_data, requires_grad=True)
+                    ctx.output_var = ctx.x * ctx.y
+                return ctx.output_var[:]
```

Comment:
what happens if you return ctx.output_var?

Reply:
Then the returned value will have the backward be Reenter.backward instead of MulBackward. This test is weird.

Reply:
Don't you have to detach it? I'm not sure what's going on in here

Reply:
Yeah, `return ctx.output_var.detach()` is probably better than `return ctx.output_var[:]` although it ends up having the same effect in this test case.